v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/schema/ISchemaOverridesParser.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.schema;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+public interface ISchemaOverridesParser {
+
+  /**
+   * Gets the spanner table name given the source table name, or source table name if no override is
+   * configured.
+   *
+   * @param sourceTableName The source table name
+   * @return The overridden spanner table name
+   */
+  String getTableOverrideOrDefault(String sourceTableName);
```

Comment:
nit: the purpose of this method is to return the name to be used in spanner. can we just call this getSpannerTableName ?

In top level interfaces it is generally better to have a deterministic function name rather than have or clauses to abstract out the logic.

Reply:
These type of methods are quite common in Java. My implementation here is just an extension of an already existing concept - 

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html#getOrDefault(java.lang.Object,V)

I am not particular about the naming here though, so if you think it is confusion, I can change it.

Reply:
Updated the name.